components/org.wso2.carbon.identity.api.server.secret.management/org.wso2.carbon.identity.api.server.secret.management.v1/src/main/resources/secretManager.yaml

```
@@ -0,0 +1,656 @@
+          style: simple
+          explode: false
+          schema:
+            type: string
+        - name: name
+          in: path
+          description: name of the secret
+          required: true
+          style: simple
+          explode: false
+          schema:
+            type: string
+      responses:
+        "204":
+          description: No Content
+        "400":
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "401":
+          description: Unauthorized
+        "403":
+          description: Forbidden
+        "404":
+          description: Not Found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "405":
+          description: Method Not Allowed.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        "500":
+          description: Server Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+components:
+  schemas:
+    Error:
+      type: object
+      properties:
+        code:
+          type: string
+          example: NSM-00000
+        message:
+          type: string
+          example: Some Error Message
+        description:
+          type: string
+          example: Some Error Description
+        traceId:
+          type: string
+          example: e0fbcfeb-3617-43c4-8dd0-7b7d38e13047
+    secretAdd:
+      required:
+        - name
+        - value
+      type: object
+      properties:
+        name:
+          type: string
+          example: choreo-riskScore
+        value:
+          type: string
+          example: bgtehaldhjdevch
+        description:
+          type: string
+          example: Some Error Description
+
+    Secret:
+      required:
+        - secretId
+        - secretName
+        - lastModified
+        - created
+        - type
+      type: object
+      properties:
+        secretId:
+          type: string
+          example: 30103923-923c-485f-a8f9-606398
+        secretName:
+          type: string
+          example: choreo-riskScore
+        created:
+          type: string
+          example: 2021.07.22
+        lastModified:
+          type: string
+          example: 2021.07.22
+        type:
+          type: string
+          example: adaptive
+        description:
+          type: string
+          example: sample_description
+
+    SecretUpdateRequest:
+      required:
+        - value
+      type: object
+      properties:
+        value:
+          type: string
+          example: bgtehaldhjdevch
+        description:
+          type: string
+          example: sample_description
+
+    SecretsList:
+      type: array
+      items:
+        $ref: '#/components/schemas/Secret'
+
+    SecretTypeAdd:
```

```suggestion
    SecretTypeAddRequest:
```
Same as secretAddRequest

---

resolved